core/src/system_monitor_service.rs

```diff
@@ -690,6 +695,112 @@ impl SystemMonitorService {
         })
     }
 
+    /*
+    fn string_append_u32_as_char(&mut string: String, val: u32)
+    {
+        let mut v = val;
+
+        for _i in 0..4 {
+            string.push(v as u8 as char);
+            v >>= 8;
+        };
+    }*/
+
+    #[cfg(any(target_arch = "x86_64", target_arch = "x86_64"))]
+    unsafe fn report_cpuid_values() {
+        let cpuid_0 = __cpuid(0);
+
+        let max_leaf = cpuid_0.eax;
+        let max_subleaf_7 = if 7 <= max_leaf {
+            __get_cpuid_max(7).1
+        } else {
+            0
+        };
+
+        let mut manufacturer = String::from("");
+        for i in 0..4 {
+            manufacturer.push((cpuid_0.ebx >> (i * 8)) as u8 as char);
+        }
+        for i in 0..4 {
+            manufacturer.push((cpuid_0.edx >> (i * 8)) as u8 as char);
+        }
+        for i in 0..4 {
+            manufacturer.push((cpuid_0.ecx >> (i * 8)) as u8 as char);
+        }
+
+        let cpuid_1 = if 1 <= max_leaf {
+            __cpuid(1)
+        } else {
+            CpuidResult {
+                eax: 0,
+                ebx: 0,
+                ecx: 0,
+                edx: 0,
+            }
+        };
+        let cpuid_2 = if 2 <= max_leaf {
+            __cpuid(2)
+        } else {
+            CpuidResult {
+                eax: 0,
+                ebx: 0,
+                ecx: 0,
+                edx: 0,
+            }
+        };
+        let cpuid_4 = if 4 <= max_leaf {
+            __cpuid(4)
+        } else {
+            CpuidResult {
+                eax: 0,
+                ebx: 0,
+                ecx: 0,
+                edx: 0,
+            }
+        };
+        let cpuid_7_0 = if 7 <= max_leaf {
+            __cpuid_count(7, 0)
+        } else {
+            CpuidResult {
+                eax: 0,
+                ebx: 0,
+                ecx: 0,
+                edx: 0,
+            }
+        };
+        let cpuid_7_1 = if 7 <= max_leaf && 1 <= max_subleaf_7 {
+            __cpuid_count(7, 1)
+        } else {
+            CpuidResult {
+                eax: 0,
+                ebx: 0,
+                ecx: 0,
+                edx: 0,
+            }
+        };
+
+        datapoint_info!(
+            "cpuid-values",
+            ("manufacturer_id", manufacturer, String),
+            ("cpuid_1_eax", cpuid_1.eax as i64, i64),
```

Comment:
prefer `i64::from` throughout
```suggestion
            ("cpuid_1_eax", i64::from(cpuid_1.eax), i64),
```

Reply:
I just copied the style from the existing "report_cpu_stats" function. Should I update that one as well?

Reply:
sure, but not in this PR.  keep it isolated to the intended changes and refrain from cleanup creep.

fwiw, `From` and `TryFrom` should be preferred over `as` in nearly all cases. `as` makes the code less explicit and doesn't enforce compile-time downcast rules and silently aliases values that don't fit smaller target types

Reply:
I checked on it, and that one is converting u64 to i64, which can't be done with from() (but can with "as", as you noted), so I'm leaving it alone.

Reply:
isn't `CpuidResult::eax` a `u32`?

Reply:
Sorry, I wasn't clear. I changed all the "as" in this function to "from", as you suggested. For the OTHER function, the pre-existing one, it uses "as" and I left that alone, because that can't be changed to "from".